assets/js/modules/analytics-4/datastore/audience-settings.js

```diff
@@ -0,0 +1,278 @@
+	 * Sets the audience segmentation setup completed by.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {number} audienceSegmentationSetupCompletedBy Audience segmentation setup completed by.
+	 * @return {Object} Redux-style action.
+	 */
+	setAudienceSegmentationSetupCompletedBy(
+		audienceSegmentationSetupCompletedBy
+	) {
+		// Should be integer.
+		invariant(
+			typeof audienceSegmentationSetupCompletedBy === 'number',
+			'Audience segmentation setup completed by should be an integer.'
+		);
+
+		return {
+			type: SET_AUDIENCE_SEGMENTATION_SETUP_COMPLETED_BY,
+			payload: { audienceSegmentationSetupCompletedBy },
+		};
+	},
+
+	/**
+	 * Saves the audience settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} settings Audience settings to save.
+	 * @return {Object} Object with `response` and `error`.
+	 */
+	saveAudienceSettings: createValidatedAction(
+		( settings ) => {
+			validateAudienceSettings( settings );
+		},
+		function* ( settings ) {
+			yield clearError( 'saveAudienceSettings', [] );
+
+			const { response, error } =
+				yield fetchSaveAudienceSettingsStore.actions.fetchSaveAudienceSettings(
+					settings
+				);
+
+			if ( error ) {
+				yield receiveError( error, 'saveUserAudienceSettings', [] );
+			}
+
+			return { response, error };
+		}
+	),
+};
+
+const baseResolvers = {
+	*getAvailableAudiences() {
+		const registry = yield commonActions.getRegistry();
+		const { select, dispatch } = registry;
+
+		const audiences = select( MODULES_ANALYTICS_4 ).getAvailableAudiences();
+
+		if ( audiences === null ) {
+			const { response, error } = yield commonActions.await(
+				dispatch( MODULES_ANALYTICS_4 ).syncAvailableAudiences()
+			);
+
+			if ( error ) {
+				return { error };
+			}
+
+			dispatch( MODULES_ANALYTICS_4 ).setAvailableAudiences( response );
+		}
+	},
+};
+
+const baseReducer = ( state, { type, payload } ) => {
+	switch ( type ) {
+		case SET_AVAILABLE_AUDIENCES: {
+			const { availableAudiences } = payload;
+
+			return {
+				...state,
+				audienceSettings: {
+					...state.audienceSettings,
+					availableAudiences,
+				},
+			};
+		}
+
+		case SET_AUDIENCE_SEGMENTATION_SETUP_COMPLETED_BY: {
+			const { audienceSegmentationSetupCompletedBy } = payload;
+
+			return {
+				...state,
+				audienceSettings: {
+					...state.audienceSettings,
+					audienceSegmentationSetupCompletedBy,
+				},
+			};
+		}
+
+		default: {
+			return state;
+		}
+	}
+};
+
+const baseSelectors = {
+	/**
+	 * Gets the available audiences.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(Array|null)} Available audiences, or `null` if not loaded.
+	 */
+	getAvailableAudiences( state ) {
+		return state.audienceSettings?.availableAudiences || null;
+	},
+	/**
+	 * Gets the audience segmentation setup completed by.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(number|null)} Audience segmentation setup completed by, or `null` if not loaded.
```

Comment:
As above, plus it should return `undefined` if not loaded:

```suggestion
	 * @return {(number|null)} ID for the user who set up Audience Segmentation, or `undefined` if not loaded.
```